nistats/tests/test_glm_reporter/glm_reporter_examples/second_level_models/report_oasis.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+import nibabel
+import pandas as pd
+import numpy as np
+
+from nilearn import datasets
+from nilearn.image import resample_to_img
+
+from nistats._utils.testing import make_reporter_example_result_path
+from nistats.reporting import generate_subject_stats_report
+from nistats.second_level_model import SecondLevelModel
+from nistats.thresholding import map_threshold
+
+
+def make_zmaps(model, contrasts):
+    """ Given a model and contrasts, return the corresponding z-maps"""
+    z_maps = {}
+    for index, (contrast_id, contrast_val) in enumerate(contrasts.items()):
+        z_maps[contrast_id] = model.compute_contrast(
+            contrast_val, output_type='z_score')
+    return z_maps
+
+def prepare_data(oasis_dataset):
+    gray_matter_map_filenames = oasis_dataset.gray_matter_maps
+    # Get a mask image: A mask of the  cortex of the ICBM template
+    gm_mask = datasets.fetch_icbm152_brain_gm_mask()
+    # Resample the images, since this mask has a different resolution
+    mask_img = resample_to_img(
+        gm_mask, gray_matter_map_filenames[0], interpolation='nearest')
+    return mask_img, gray_matter_map_filenames
+
+
+def make_stats_model(mask_img, gray_matter_map_filenames, design_matrix):
+    # Specify and fit the second-level model when loading the data, we
+    # smooth a little bit to improve statistical behavior
+    second_level_model = SecondLevelModel(smoothing_fwhm=2.0, mask=mask_img)
+    second_level_model.fit(gray_matter_map_filenames,
+                           design_matrix=design_matrix)
+    # Estimate the contrast is very simple. We can just provide the column
+    # name of the design matrix.
+    z_map = second_level_model.compute_contrast(second_level_contrast=[1, 0, 0],
+                                                output_type='z_score')
+    # We threshold the second level contrast at uncorrected p < 0.001 and plot it.
+    # First compute the threshold.
+    _, threshold = map_threshold(
+        z_map, level=.05, height_control='fdr')
+    # Can also study the effect of sex: compute the stat, compute the
+    # threshold, plot the map
+    z_map = second_level_model.compute_contrast(second_level_contrast='sex',
+                                                output_type='z_score')
+    _, threshold = map_threshold(
+        z_map, level=.05, height_control='fdr')
+    z_maps = {'age': z_map}
```

Comment:
The z_map is actually for the regressor 'sex' (see line 50 above). 
I think it would be nice to modify the example to return two z-maps, one for 'age' and one for 'sex'. 
Even more generally, I think it would be nice to have a function that, given a (FS)LM object with a design matrix (or several design matrices that have the same column names), would compute the effect-sizes and z-maps associated to each regressor.